### PR TITLE
Move test of sim_red_data over from hera_cal.redcal

### PR DESCRIPTION
**Pull request by jsdillon**
This will help us finally close https://github.com/HERA-Team/hera_cal/issues/379

### CODE REVIEW

File: hera_sim/vis.py [hunk 1]
```diff
@@ -162,7 +163,7 @@ def hmap_to_I(h):
 #     return gains, true_vis, data
 
 
-# XXX from hera_cal.redcal
+# from hera_cal.redcal
 def sim_red_data(reds, gains=None, shape=(10, 10), gain_scatter=0.1):
     """
     Simulate noise-free random but redundant (up to differing gains) visibilities.
```
Comment (steven-murray) [inline hera_sim/vis.py]: Are these noise-free? I see you call `noise.white_noise` in the body of the function.

File: hera_sim/tests/test_vis.py [hunk 2]
```diff
@@ -94,6 +94,111 @@ def test_values(self):
             v[:, 0, 1], 1 + np.exp(-2j * np.pi * np.sqrt(0.5)), 7
         )
 
+class TestSimRedData(unittest.TestCase):
```
Comment (steven-murray) [inline hera_sim/tests/test_vis.py]: Other than knowing that this is testing `sim_red_data`, I have no idea what you are testing here. Could you supply a few higher-level comments?

File: hera_sim/vis.py (hunk 1 above)
Comment (jsdillon) [reply hera_sim/vis.py]: Yes. The visibilities and gains are random, but the baselines that should be redundant are perfectly redundant. I'll clarify that it's thermal noise free.

File: hera_sim/tests/test_vis.py (hunk 2 above)
Comment (jsdillon) [reply hera_sim/tests/test_vis.py]: Done.

**Comment by jsdillon** [comment]
@steven-murray I think this is ready for your review again.

**Pull request merged**